### PR TITLE
Goupil G2 computer support added.

### PR DESCRIPTION
**Pull request by jfdelnero**

### CODE REVIEW

File: src/mame/drivers/goupil.cpp [hunk 1]
```diff
@@ -96,6 +98,34 @@ class goupil_g1_state : public driver_device
...
+	{
+	}
+
+	MC6845_UPDATE_ROW(crtc_update_row);
+	MC6845_ON_UPDATE_ADDR_CHANGED(crtc_update_addr_changed);
+
+	DECLARE_READ8_MEMBER(visu24x80_ram_r);
+	DECLARE_WRITE8_MEMBER(visu24x80_ram_w);
+
+	required_device<palette_device> m_palette;
```
Comment (cuavas) [inline src/mame/drivers/goupil.cpp]: Is there a reason all these members need to be public?  If not, it's better to make them protected/private.

File: src/mame/drivers/goupil.cpp [hunk 2]
```diff
@@ -480,32 +585,116 @@ static MACHINE_CONFIG_START( goupil_g1 )
...
+
+ROM_END
+
 /* Driver */
 
-//    YEAR  NAME      PARENT  COMPAT  MACHINE    INPUT      CLASS            INIT  COMPANY  FULLNAME       FLAGS
-COMP( 1979, goupilg1, 0,      0,      goupil_g1, goupil_g1, goupil_g1_state, 0,    "SMT",   "Goupil G1",   MACHINE_IS_SKELETON )
+/*    YEAR   NAME   PARENT  COMPAT   MACHINE    INPUT  CLASS           INIT    COMPANY   FULLNAME       FLAGS */
+COMP( 1979, goupilg1,   0,   0,      goupil_g1,  goupil_g1,goupil_g1_state,   0,     "SMT", "Goupil G1",    MACHINE_TYPE_COMPUTER | MACHINE_IS_INCOMPLETE | MACHINE_NO_SOUND )
+COMP( 1981, goupilg2,   0,   0,      goupil_g2,  goupil_g1,goupil_g2_state,   0,     "SMT", "Goupil G2",    MACHINE_TYPE_COMPUTER | MACHINE_IS_INCOMPLETE | MACHINE_NO_SOUND )
```
Comment (cuavas) [inline src/mame/drivers/goupil.cpp]: You don't put `MACHINE_TYPE_COMPUTER` in the flags - that's implied by using the `COMP` macro.  Also, are these actual incomplete prototypes?  `MACHINE_IS_INCOMPLETE` is not for incomplete emulation, it's for emulation of a prototype that's considered incomplete.

File: src/mame/drivers/goupil.cpp (hunk 1 above)
Comment (jfdelnero) [inline src/mame/drivers/goupil.cpp]: done.

File: src/mame/drivers/goupil.cpp (hunk 2 above)
Comment (jfdelnero) [inline src/mame/drivers/goupil.cpp]: fixed.

**Pull request merged**